src/controller.cpp

```diff
@@ -291,6 +291,10 @@ void Controller::activateInternal() {
 void Controller::activateNext() {
   MozillaVPN* vpn = MozillaVPN::instance();
   const Device* device = vpn->deviceModel()->currentDevice(vpn->keys());
+  if (device == nullptr) {
+    vpn->errorHandle(ErrorHandler::NoCurrentDeviceError);
```

Comment:
Instead of adding a new error here, what about adding:
```
TaskScheduler::scheduleTask(new TaskFunction([this]() {
    if (!modelsInitialized()) {
      logger.error() << "Failed to complete the key regeneration";
      errorHandle(ErrorHandler::RemoteServiceError);
      setUserState(UserNotAuthenticated);
      return;
    }
});
```

here: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/mozillavpn.cpp#L1680

Reply:
We can do that! But i still think we should check that it's not a `nullptr` aynway.